Reword the metadata gudelines (#283)
EVERSE-ResearchSoftware/RSQKit: pages/contributing/metadata_guidelines.md

summary: Descriptions of different metadata schemas used to describe various resources and pages in RSQKit.
---

This page explains the metadata that can be used to describe various resources and pages in RSQKit. 
These guidelines have been adapted from the ELIXIR Toolkit theme's [page metadata guide](https://elixir-belgium.github.io/elixir-toolkit-theme/page_mechanics)
and [tools and resources guide](https://elixir-belgium.github.io/elixir-toolkit-theme/resource_table).

## Contributor metadata

Contributors are listed in the [contributors file (`_data/CONTRIBUTORS.yml`)](https://github.com/EVERSE-ResearchSoftware/RSQKit/blob/main/_data/CONTRIBUTORS.yml)
starting with the contributor's full name, followed by the following optional metadata/attributes:

* `git`: contributor's GitHub id (serving as an identifier together with contributor's full name).
* `orcid`: contributor's ORCID id.

> * `role`: role of the contributor in the RSQKit (e.g. `editor`).

Reviewer: perhaps we need to say for any other role than editor this can be blank

Author: I say that all parameters apart from the name are optional but I get your point - perhaps it is better to emphasise that.

Author: That's another thing I wanted to tlk to you about - add more explicit roles - contributor, author, editor etc. to make it easier to explain what different contribution type are there. Having multiple roles would be great but the ETT does not support this.

> * `role`: role of the contributor in the RSQKit (e.g. `editor`).

Author: ```suggestion
* `role`: role of the contributor in the RSQKit (at the moment, we only support the role `editor`, but more may be added in the future; leave blank if the role is not `editor`).
```

* `contributors`: a list of contributors that authored or contributed significantly to the page in some way (e.g. via discussions). Each contributor must also be listed in the `_data/CONTRIBUTORS.yml` repository file.
* `search_exclude`: a boolean value indicating if the page should be excluded from search results. Default: `false`.
* `sitemap`: a boolean value indicating if the page should appear in the `sitemap.xml`. Default: `true`.
* `no_robots`: a boolean value indicating if the page should not end up in the search results of Google or any other search engines. Default: `false`.
* `hide_sidebar`: a boolean value indicating if the page should not appear in the sidebar. Default: `false`.
* `custom_editme`: specify an alternative file/link when clicking on the `edit-me` button.
* `keywords`: a list of lowercase keywords that can be used to find the page using the search facility of RSQKit.
* `sidebar`: name of the left-hand side navigation sidebar that should be displayed for the page. Default sidebar: `main`. The sidebar file `<SIDEBAR_NAME.md>` named after the sidebar must exist under `_data/sidebars/` in the repository.

> * `toc`: a boolean value indicating if a table of contents should be included at the top of the page. Default: `false`.

Reviewer: ```suggestion
* `toc`: a boolean value indicating if a table of contents should be included at the top right of the page. Default: `false`.
```

* `page_id`: unique identifier of the page, usually a shortened version of the page title (with words separated with underscores or dashes and spaces avoided). This identifier is used in `related_pages` parameter to list pages related to this page. 
* `datatable`: a boolean value indicating the activation of the pagination, sorting and searching in tabular representations of pages.
* `related_pages`: a list of `page_id`s that are related to this page and will appear under "Related pages" section on the page, grouped by page type.

> * `page_citation`: When set to true, it will cause the citation section for the page to be generated in the format: "<author names>. <page title>. <site domain>. <page URL>. <date accessed>".

Reviewer: What's the default - I thought it was true - also in the netlify I get format "...." rather than the text - maybe it's different if built locally

> * `page_citation`: When set to true, it will cause the citation section for the page to be generated in the format: "<author names>. <page title>. <site domain>. <page URL>. <date accessed>".

Author: ```suggestion
* `page_citation`: When set to `true`, it will cause the citation section for the page to be generated in the format: "<author names>. <page title>. <site domain>. <page URL>. <date accessed>". Defaults to `true` for task pages; `false` for other page types.
```

* `name`: name of the tool or resource.
* `description`: a short description of the tool or resource.
* `id`: identifier used to refer to the tool or resource in pages.
* `url`: URL of the tool or resource
* `quality_indicator`:  research software quality indicators associated with this tool. We have not settled 100% on the format of attribute yet - this is still work in progress. 
Most likely it will be a list of research software quality indicators (defined in the [quality indicators data file (`_data/quality_indicators.yml`)](https://github.com/EVERSE-ResearchSoftware/RSQKit/blob/main/_data/quality_indicators.yml)).

An example of a tool definition is given below:

```yml
- description: > 
    The System Package Data Exchange (SPDX) License List is a list of commonly found licenses 
    and exceptions used in free and open or collaborative software, data, hardware, or 
    documentation, and is an integral part of the SPDX Specification.
  id: spdx
  name: SPDX License List
  url: https://spdx.org/licenses/
```

Pages in RSQKit can include references to tools and resources. Such references will appear visually different on a page - 

> with a little wrench icon and a pop-up window which shows up on hover over and includes the tool/resource description.

Reviewer: ```suggestion
with a little wrench icon and a pop-up window which shows up on hover over and includes the tool/resource description and a website link.
```

To refer to a tool or a resource in a page, use the following syntax:

{% raw %}
```
{% tool "tool-id" %}
```
{% endraw %}

> All tools and resources mentioned on a page will be listed in a table at the bottom of the page as well as under "All Tools and Resources" page. 

Reviewer: ```suggestion
All tools and resources mentioned on a page will be listed in a table at the bottom of the page as well as under "[All Tools and Resources](all_tools_and_resources)" page. 
```
